tests/README.md

```diff
@@ -1,25 +1,242 @@
+The dependencies for the integration tests are collected and organized in
+dependency files in the `./requirements` folder, and separated into
+
+
+> These requirement files are using the *hashicorp/bionic64* *Vagrant* image is
+> as the basis for the requirements files generated.
+
+| Debian                 | Alpine                 | Python                    |
+| :-------------:        | :-------------:        | :-----:                   |
+| *deb-requirements.txt* | *apt-requirements.txt* | *python-requirements.txt* |
+
+## Installing Dependencies
+
+### Debian
+
+```bash
+$ apt install -yyq $(cat requirements/deb-requirements.txt)
+```
+
+### Alpine Linux
+
+```bash
+$ apk --update add -yyq $(cat requirements/apk-requirements.txt)
+```
+
+### Python2 and Python3
+
+```bash
+$ pip  install -r requirements/python-requirements.txt
+$ pip3 install -r requirements/python-requirements.txt
+```
+
+
+-------------------------------------------------------------------------------
+
+
+## Running the Tests Locally
+
+> The tests can be run locally without any further involvement as long as all
+> the dependencies have been installed and are at the correct version. However,
+> managing dependencies, especially with Python can be a hassle. Therefore it is
+> recommended to add a virtual Python environment to isolate the dependencies
+> needed for running the integration tests.
+
+### Isolating Python Dependencies
+
+In order to avoid dependency mismanagement due to Python packages differing from
+one test environment to the other, it is recommended to use a python virtual
+environment. The "de-facto" standard is
+[virtualenv](https://packaging.python.org/guides/installing-using-pip-and-virtual-environments/).
+
+#### Install
+
+```bash
+$ pip install virtualenv
+```
+
+#### Initialize the Virtual Environment
+
+```bash
+$ cd integration-dir/tests
+$ virtualenv <name-of-virtualenv-folder>
+```
+
+##### (Optional) -- Select Python Version
+```bash
+$ virtualenv -p /usr/bin/python2.7 <name-of-virtualenv-folder>
+```
+
+#### Activate the Virtual Environment
+
+```bash
+$ source <name-of-virtualenv-folder>/bin/activate
+```
+
+This now means that you have a clean python environment, and no packages you
+have previously installed outside of this virtual environment will be
+discoverable by Python.
+
+Once you are finished the virtual environment is deactivated with
+
+```bash
+$ deactivate
+```
+
+#### Install the Python Dependencies from the Requirements File
+
+```bash
+$ pip install -r requirements/python-requirements.txt
+```
+
+This will install the requirements needed to run the tests.
+
+### Running the Tests
+
+Next, run the tests with the `run.sh` script.
 
-####Known issues
+```bash
+$ ./run.sh
+```
 
-Since we attempting to SSH into the virtual mender device, before the OS is up and running, you may see errors such as:
+**NOTE**: This is dependent upon having a functioning Docker environment, and being
+logged in to `registry.mender.io`.
```

Comment:
Mmmm. You should be able to run the Open Source tests without logging in. So I would rephrase it a bit to show that it is not a requirement

Reply:
In general the run-script has a fair few options, that are not really documented. Originally I was thinking about adding it to the readme, but a usage flag to run.sh would probably be better. Thoughts?

Reply:
And, also, it seems not all Enterprise tests are collected in the backend tests yet, which means that now `run.sh` will run (amongst others), `TestMultiTenancyEnterprise`. Which means it is required I think

Reply:
Yes it is better, but now it is inconsistent. For the open source you used the `SPECIFIC_INTEGRATION_TEST` method while for enterprise you use the `-k` method. We should align and IMO the least hacky one is the `-k`

Reply:
Yes, I agree. The reason though, is that `./run.sh -k "not Enterprise" does not work :p

So if we are to align them, then the env var should be used in both cases.

Reply:
I see. Let's use the env var then